src/System.IO.Pipelines/System/IO/Pipelines/PipeWriter.cs

```diff
@@ -33,5 +36,24 @@ public abstract class PipeWriter: IOutput
         public abstract void Advance(int bytes);
         public abstract Memory<byte> GetMemory(int minimumLength = 0);
         public abstract Span<byte> GetSpan(int minimumLength = 0);
+
+        public virtual Task WriteAsync(ReadOnlyMemory<byte> source)
```

Comment:
Not entirely sure about this, maybe it needs to return `ValueTask<FlushResult>`

Reply:
Yea, lets do ValueAwaiter<FlushResult>

Reply:
Fixed